Lib/importlib/_bootstrap.py

```diff
@@ -957,16 +959,17 @@ def _find_and_load_unlocked(name, import_):
 
 def _find_and_load(name, import_):
     """Find and load the module."""
-    _imp.acquire_lock()
-    if name not in sys.modules:
-        with _ModuleLockManager(name):
+    with _ModuleLockManager(name):
+        try:
+            module = sys.modules[name]
+        except KeyError:
```

Comment:
KeyError is raised in common case (_find_and_load is called from accelerated C code when name not in sys.modules). Catching exceptions in Python code is slow. If you want to avoid KeyError in the following code, it would be better to use `sys.modules.get(name, sentinel)`.